components/idp-mgt/org.wso2.carbon.idp.mgt.ui/src/main/java/org/wso2/carbon/idp/mgt/ui/util/IdPManagementUIUtil.java

```diff
@@ -961,10 +985,25 @@ private static void buildBasicInformation(IdentityProvider fedIdp, Map<String, S
 
         // if there is no new certificate and not a delete - use the old one.
         if (oldCertFile != null && certFile == null
-                && (deletePublicCert == null || ("false").equals(deletePublicCert))) {
+                && (deletePublicCert == null || !Boolean.parseBoolean(deletePublicCert))) {
```

Comment:
Boolean.parseBoolean() will return false for a null string. So no need to check for a null string specifically.

Reply:
Fixed with https://github.com/wso2/carbon-identity-framework/pull/1704/commits/cdea1a45f0fde05acfbca6400cf8c41db1ee6493